src/main/kotlin/no/nav/syfo/kartleggingssporsmal/domain/Oppfolgingstilfelle.kt

```diff
@@ -1,39 +1,135 @@
-    val tilfelleStart: LocalDate,
-    val tilfelleEnd: LocalDate,
-    val antallSykedager: Int?,
-    val dodsdato: LocalDate?,
-    val isArbeidstakerAtTilfelleEnd: Boolean,
-    val virksomhetsnummerList: List<String>,
+sealed class Oppfolgingstilfelle(
+    open val isArbeidstakerAtTilfelleEnd: Boolean,
+    open val tilfelleStart: LocalDate,
+    open val tilfelleEnd: LocalDate,
+    open val personident: Personident,
+    open val antallSykedager: Int?,
+    open val dodsdato: LocalDate?,
+    open val virksomhetsnummerList: List<String>,
 ) {
 
-    fun hasTilfelleWithEndMoreThanThirtyDaysAgo(): Boolean = this.tilfelleEnd isMoreThanDaysAgo 30
+    data class OppfolgingstilfelleFromKafka internal constructor(
+        override val personident: Personident,
+        override val tilfelleStart: LocalDate,
+        override val tilfelleEnd: LocalDate,
+        override val antallSykedager: Int?,
+        override val dodsdato: LocalDate?,
+        override val isArbeidstakerAtTilfelleEnd: Boolean,
+        override val virksomhetsnummerList: List<String>,
+        val uuid: UUID,
+        val tilfelleGenerert: OffsetDateTime,
+        val tilfelleBitReferanseUuid: UUID,
+    ) : Oppfolgingstilfelle(
+        isArbeidstakerAtTilfelleEnd,
+        tilfelleStart,
+        tilfelleEnd,
+        personident,
+        antallSykedager,
+        dodsdato,
+        virksomhetsnummerList,
+    ) {
+        fun hasTilfelleWithEndMoreThanThirtyDaysAgo(): Boolean = this.tilfelleEnd isMoreThanDaysAgo 30
 
-    fun isDod() = dodsdato != null
+        fun isDod() = dodsdato != null
 
-    fun durationInDays(): Long {
-        return if (this.antallSykedager != null) {
-            antallSykedager.toLong()
-        } else {
-            ChronoUnit.DAYS.between(this.tilfelleStart, this.tilfelleEnd) + 1
+        fun durationInDays(): Long {
+            return if (this.antallSykedager != null) {
+                antallSykedager.toLong()
+            } else {
+                ChronoUnit.DAYS.between(this.tilfelleStart, this.tilfelleEnd) + 1
+            }
         }
+
+        fun datoInsideCurrentTilfelle(dato: LocalDate): Boolean =
+            dato isAfterOrEqual this.tilfelleStart && dato isBeforeOrEqual this.tilfelleEnd
+    }
+
+    data class OppfolgingstilfelleFromApi internal constructor(
+        override val personident: Personident,
+        override val tilfelleStart: LocalDate,
+        override val tilfelleEnd: LocalDate,
+        override val antallSykedager: Int?,
+        override val dodsdato: LocalDate?,
+        override val isArbeidstakerAtTilfelleEnd: Boolean,
+        override val virksomhetsnummerList: List<String>,
+    ) : Oppfolgingstilfelle(
+        isArbeidstakerAtTilfelleEnd,
+        tilfelleStart,
+        tilfelleEnd,
+        personident,
+        antallSykedager,
+        dodsdato,
+        virksomhetsnummerList,
+    ) {
+        fun isActive(): Boolean = LocalDate.now().isBeforeOrEqual(this.tilfelleEnd)
```

Comment:
API-oppslaget skal brukes for å sjekke om personen fortsatt er sykmeldt på stoppunkt-tidspunktet. Hvis de er det, så blir de kandidat (gitt at andre regler går gjennom).